### PR TITLE
Updated toyform with the correct syntax

### PR DESCRIPTION
**Pull request by Halezmarie**

### CODE REVIEW

File: src/components/ToyForm.js [hunk 1]
```diff
@@ -8,12 +8,12 @@ const ToyForm = ({ toy }) => {
...
     name: '',
     description: '',
-    image_url: '',
+    imageUrl: '',
   });
 
   const handleChange = (event) => {
     setForm({
-      ...form,
+      form,
```
Comment (ssteele-godaddy) [inline src/components/ToyForm.js:16]: `form` initializes as:
```
{ name: '', description: '', imageUrl: '' }
```

When handleChange hits (assuming I typed "My Name" into name field), it's going to set form to this:
```
{ form: { name: '', description: '', imageUrl: '' }, name: 'My name' }
```
Then, if I change "My name" to "Your name", it's going to look like this:
```
{ form: { form: { name: '', description: '', imageUrl: ''}, name: 'My name' }, name: 'Your name' }
```

File: src/components/ToyForm.js [hunk 2]
```diff
@@ -22,13 +22,15 @@ const ToyForm = ({ toy }) => {
     event.preventDefault();
     dispatch(createToy(form, toy));
     setForm({
-      ...form,
+      form,
```
Comment (ssteele-godaddy) [inline src/components/ToyForm.js:25]: Please see my original comments (and see above as well).  Passing form here is going to create an object:
```
{ form: { name: '', description: '', imageUrl: '' }, name: '', description: '', imageUrl: '' }
```

..then progressively get larger each time.

Comment (ssteele-godaddy) [reply src/components/ToyForm.js:25]: https://jsfiddle.net/rh9txdjn/

File: src/components/ToyForm.js (hunk 1 above)
Comment (ssteele-godaddy) [reply src/components/ToyForm.js:16]: https://jsfiddle.net/zg4nu6ok/

**Review by ssteele-godaddy — CHANGES_REQUESTED** [review]

File: src/components/ToyForm.js (hunk 2 above)
Comment (Halezmarie) [reply src/components/ToyForm.js]: Hey Shawn! I went ahead and updated this form. I hope you like the changes. I also got the `imageUrl `to work for the front-end and the back-end. Is that code okay? Thanks!

File: src/components/ToyForm.js [hunk 3]
```diff
@@ -20,15 +20,17 @@ const ToyForm = ({ toy }) => {
 
   const handleSubmit = (event) => {
     event.preventDefault();
-    dispatch(createToy(form, toy));
+    const { imageUrl, ...rest } = form;
+    dispatch(createToy({ image_url: imageUrl, ...rest }, toy));
```
Comment (ssteele-godaddy) [inline src/components/ToyForm.js:24]: This is looking better, but when handing things like tweaking field names for sending via an API call, think about where the responsibility lies.  Is it the components responsibility to care what the data looks like when making the API call?  Probably not, it shouldn't care about the API at all.  It feels like responsibility should be where the API call is being made-- the `createToy`action.

**Review by ssteele-godaddy — CHANGES_REQUESTED** [review]

**Review by ssteele-godaddy — APPROVED** [review]

**Pull request merged**